sycl/source/detail/scheduler/scheduler2.cpp

```diff
@@ -21,7 +21,16 @@ namespace sycl {
 namespace detail {
 
 void Scheduler::waitForRecordToFinish(GraphBuilder::MemObjRecord *Record) {
-  for (Command *Cmd : Record->MLeafs) {
+  for (Command *Cmd : Record->MReadLeafs) {
+    Command *FailedCommand = GraphProcessor::enqueueCommand(Cmd);
+    if (FailedCommand) {
+      // TODO: What is the best way to handle failed command in Scheduler d'tor?
+      assert(!FailedCommand && "Command failed to enqueue");
```

Comment:
I'd like to see in the error message that we failed to enqueue read command here.

I'd like to see in the error message that we failed to enqueue write command below.